### PR TITLE
fix(clerk-js): UI fixes on <UserProfile /> to match new design

### PR DESCRIPTION
**Pull request by marcelscruz**
## Description

Fixes the issues listed on [this ticket](https://linear.app/clerk/issue/SDK-1158/[marcel-testing]-user-profile-ui-fixes).

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Comment by marcelscruz** [comment]
!preview

**Comment by marcelscruz** [comment]
!preview

**Comment by marcelscruz** [comment]
!preview

**Comment by octoper** [comment]
!preview

**Comment by clerk-cookie** [comment]
Hey @octoper, your preview is currently deploying and should be available shortly.

| Status | Preview | Updated (UTC) |
| :-- | :-- | :-- |
| :cook: _Deploying..._ | [Inspect](https://javascript-preview-606ra5vzf.clerkstage.dev) | Jan 12, 2024 12:01 PM |

File: packages/clerk-js/src/ui/components/UserProfile/MfaBackupCodeList.tsx [hunk 1]
```diff
@@ -45,15 +45,13 @@ export const MfaBackupCodeList = (props: MfaBackupCodeListProps) => {
   return (
     <>
       <Col gap={1}>
-        <Text
-          localizationKey={localizationKeys('userProfile.backupCodePage.title__codelist')}
```
Comment (anagstef) [inline packages/clerk-js/src/ui/components/UserProfile/MfaBackupCodeList.tsx:49]: Is this key getting removed?

File: packages/clerk-js/src/ui/elements/Section.tsx [hunk 2]
```diff
@@ -247,6 +247,10 @@ export const SectionHeader = (props: SectionHeaderProps) => {
         variant='h3'
         elementDescriptor={textElementDescriptor}
         elementId={textElementId}
+        sx={t => ({
+          fontWeight: t.fontWeights.$medium,
```
Comment (anagstef) [inline packages/clerk-js/src/ui/elements/Section.tsx:251]: We don't use directly fontWeights. Text should be represented only by the text variants we already have in our typography.

Comment (octoper) [reply packages/clerk-js/src/ui/elements/Section.tsx]: Fixed!

File: packages/clerk-js/src/ui/components/UserProfile/MfaBackupCodeList.tsx (hunk 1 above)
Comment (octoper) [reply packages/clerk-js/src/ui/components/UserProfile/MfaBackupCodeList.tsx]: Reverted the change!

**Review by anagstef — APPROVED** [review]

**Pull request merged**